CHANGELOG.md

```diff
@@ -1,5 +1,90 @@
 # Changelog
 
+## [v0.4.0](https://github.com/delta-incubator/delta-kernel-rs/tree/v0.4.0/) (2024-10-22)
+
+[Full Changelog](https://github.com/delta-incubator/delta-kernel-rs/compare/v0.3.1...v0.4.0)
+
+**API Changes**
+
+*Breaking*
+
+1. `pub ScanResult.mask` field made private and only accessible as `ScanResult.raw_mask()` method [\#374]
+2. `ReaderFeatures` enum changes
+3. `WriterFeatures` enum changes
+4. `Error` enum changes
+5. Module moved: `mod delta_kernel::transaction` -> `mod delta_kernel::actions::set_transaction` [\#386]
+6. apply a schema to fix column names [\#331]
+7. change `default-feature` to be none [\#339]
+8. make `scan::execute` return a lazy iterator [\#340]
+9. filesystem client now always returns a sorted list [\#344]
```

Comment:
This isn't really breaking is it? There was some internal changes here, but the api contract always said it should be sorted, we were just not sorting when we needed to sometimes

Reply:
ah yep good call - i'll move to 'fixes'?